jetty-server/src/test/java/org/eclipse/jetty/server/ResponseTest.java

```diff
@@ -578,6 +578,30 @@ public void testResetContentTypeWithCharacterEncoding() throws Exception
         assertEquals("foo2/bar2;charset=utf-8", response.getContentType());
     }
 
+
+    @Test
+    public void testPrintln() throws Exception
+    {
+        Response response = getResponse();
+        Request request = response.getHttpChannel().getRequest();
+
+        SessionHandler session_handler = new SessionHandler();
+        session_handler.setServer(_server);
+        session_handler.setUsingCookies(true);
+        session_handler.start();
+        request.setSessionHandler(session_handler);
+        HttpSession session = request.getSession(true);
+
+        assertThat(session,not(nullValue()));
+        assertTrue(session.isNew());
+
+        response.getOutputStream().println("ABC");
+        response.getOutputStream().println("XYZ");
+        response.getOutputStream().close();
+        assertEquals("ABC\r\nXYZ\r\n",BufferUtil.toString(_content));
+    }
```

Comment:
Given the many ifs and elses in the implementation, we should at least cover few of them with more tests (e.g. the grow case if we have it, the multiple passes at encoding, underflows, overflows, malformed and unmappable characters.

Reply:
OK

Reply:
Added a test that does an overflow.  Coverage indicates most of the method is tested.